src/ripple/app/misc/CanonicalTXSet.cpp

```diff
@@ -76,13 +76,24 @@ bool CanonicalTXSet::Key::operator>= (Key const& rhs)const
 
 void CanonicalTXSet::insert (std::shared_ptr<STTx const> const& txn)
 {
-    uint256 effectiveAccount = mSetHash;
-
-    effectiveAccount ^= to256 (txn->getAccountID(sfAccount));
-
-    mMap.insert (std::make_pair (
-                     Key (effectiveAccount, txn->getSequence (), txn->getTransactionID ()),
-                     txn));
+    auto const& accountId = txn->getAccountID(sfAccount);
+
+    // This assumes uint256 is zero-initialized
+    uint256 effectiveAccount;
```

Comment:
The default constructor zero-initializes, and that's bad behavior - I was intending to mark the code so that it's crystal clear that we rely on this behavior. To do that, I think that `uint256 effectiveAccount = beast::zero;` is the best option - no need for a comment.